### PR TITLE
docs(examples): add `leaflet` example

### PR DESCRIPTION
**Pull request by abereghici**
Add an example of using leaflet map with remix.

### CODE REVIEW

File: examples/leaflet/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# TODO: Example app with leaflet map
```
Comment (machour) [inline examples/leaflet/README.md:1]:
```suggestion
# Example app with leaflet map
```

File: examples/leaflet/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Leaflet cannot be rendered on the server side, so we're using the `ClientOnly` component to display a skeleton instead.
+It's important to add the `.client.tsx` suffix on `Map` component. Otherwise, a similar error will be displayed:
+
+```
+Error [ERR_REQUIRE_ESM]: require() of ES Module /remix/examples/leaflet/node_modules/react-leaflet/lib/index.js from /remix/examples/leaflet/build/index.js not supported.
+```
+
+Make sure you don't import the leaflet styles from `node_modules`, but use a CDN link ([https://unpkg.com/leaflet@1.8.0/dist/leaflet.css](https://unpkg.com/leaflet@1.8.0/dist/leaflet.css)).
+The styles from `node_modules` will cause map assets issues (ex: 404 for marker icons)
```
Comment (machour) [inline examples/leaflet/README.md:30]: In my app, I solved this by hosting the icons in my `public` folder.

Could you rephrase this so that it doesn't sound like using CSS from the CDN is the *only* possible option?

File: examples/leaflet/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Related Links
+
+Link to documentation or other related examples.
```
Comment (machour) [inline examples/leaflet/README.md:34]:
```suggestion
```

File: examples/leaflet/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+This example shows how to use Leaflet with Remix.
+
+Relevant files:
+
+- [app/client-only.tsx](app/client-only.tsx)
+- [app/map.client.tsx](app/map.client.tsx)
+
+Leaflet cannot be rendered on the server side, so we're using the `ClientOnly` component to display a skeleton instead.
+It's important to add the `.client.tsx` suffix on `Map` component. Otherwise, a similar error will be displayed:
```
Comment (machour) [inline examples/leaflet/README.md:23]:
```suggestion
It's important to add the `.client.tsx` suffix to the `Map` component file name, otherwise, you will get this error:
```

File: examples/leaflet/app/root.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+              style={{ height: mapHeight, background: "#d1d1d1" }}
+            />
+          }
+        >
+          {() => <Map height={mapHeight} />}
+        </ClientOnly>
+      </body>
+    </html>
+  );
+}
```
Comment (machour) [inline examples/leaflet/app/root.tsx:54]: Can you please move the Map handling & display to a `app/routes/index.tsx` file?
People are used to at least having the index route, and I think having the whole app in `root.tsx` will confuse them.

File: examples/leaflet/package.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
+  "private": true,
+  "sideEffects": false,
+  "scripts": {
+    "build": "remix build",
+    "dev": "remix dev",
+    "start": "remix-serve build"
+  },
+  "dependencies": {
+    "@remix-run/node": "1.6.4",
```
Comment (machour) [inline examples/leaflet/package.json:10]: can you bump all remix packages to the latest version?

**Review by machour — CHANGES_REQUESTED** [review]
Thank you @abereghici !

Some feedback to address before the team validates this very welcome example.

File: examples/leaflet/README.md (hunk 2 above)
Comment (abereghici) [reply examples/leaflet/README.md]: Thanks, @machour for reviewing it! I updated the README.file. I hope now is better :D

File: examples/leaflet/app/routes/index.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { LinksFunction } from "@remix-run/node";
```
Comment (machour) [inline examples/leaflet/app/routes/index.tsx:1]:
```suggestion
import type { LinksFunction } from "@remix-run/node";
```

🤓

**Review by machour — APPROVED** [review]

**Review by chaance — APPROVED** [review]
